gdk/wizard/commands/prompt.py

```diff
@@ -0,0 +1,270 @@
+        self.parser = argparse.ArgumentParser()
+
+    def prompter(self, field, value, required, max_attempts=3):
+        """
+        Prompts the user for a value of a given field key
+
+        Parameters
+        ----------
+            field (enum): a field key of the gdk_config file to be changed
+            value (string): the current value corresponding the field key to be changed
+            required (boolean): if the field is a required field of the gdk config file
+            max_attempts (int): the maximum number of attempts to get a valid response from the user
+
+        Returns
+        -------
+            string: the value for the field key "field"
+        """
+        require = "required " if required else "OPTIONAL "
+        link = "https://docs.aws.amazon.com/greengrass/v2/developerguide/gdk-cli-configuration-file.html#gdk-config-format"
+        for attempt in range(1, max_attempts + 1):
+            parser_argument = field.value.key
+            if field == Fields.BUILD_OPTIONS:
+                parser_argument = "build_options"
+            elif field == Fields.PUBLISH_OPTIONS:
+                parser_argument = "publish_options"
+
+            args = self.parser.parse_args(
+                [
+                    f"--{parser_argument}",
+                    self.interactive_prompt(field.value.key, value, require),
+                ]
+            )
+            response = getattr(args, parser_argument).strip()
+
+            # only way customer is asked about custom_build_command if they have custom build system
+            # in which case they must specify a custom build command that is not None
+            if field == Fields.CUSTOM_BUILD_COMMAND:
+                if response == "None":
+                    print(
+                        f"Attempt {attempt}/{max_attempts}: Must Specify a custum build command.\nPlease vist: {link}"
+                    )
+                    continue
+
+            # if customer input response is the default value or the customer input response is
+            # the same as the current value, return the current value, then return the same value
+            if (
+                response == field.value.default
+                or response == value
+                or self.checker.is_valid_input(response, field)
+            ):
+                return response
+            print(
+                f"Attempt {attempt}/{max_attempts}: Invalid response. Please try again.\nPlease vist: {link}"
+            )
+
+        if field == Fields.CUSTOM_BUILD_COMMAND:
+            print(
+                "You have failed to enter a valid custom build command. Exiting wizard..."
+            )
+            # write to all current values to the config file and exit the wizard
+            # self.write_to_config_file()
+
+        print("Exceeded maximum attempts. Assuming default response.")
+        return value
+
+    def change_build_or_publish(self, build_or_publish, max_attempts=3):
```

Comment:
Why are we asking for these two together?

Reply:
Oh its not asking both questions at once. It's asking about build or publish depending on the value of build_or_publish

Reply:
I understand that nothing changes on the customer side but I don't think we need to club these two together in the code. Build and publish are two independent configs so coupling these two will make it hard for us to maintain and test the code in future.

Reply:
Got it I will separate them out